bin/strata-client/src/args.rs

```diff
@@ -57,3 +62,114 @@ pub struct Args {
     #[argh(option, description = "database retry count")]
     pub db_retry_count: Option<u16>,
 }
+
+impl Args {
+    pub fn derive_config(&self) -> Result<Config, String> {
+        let args = self.clone();
+        Ok(Config {
+            bitcoind_rpc: BitcoindConfig {
+                rpc_url: args
+                    .bitcoind_host
+                    .ok_or_else(|| "args: no bitcoin --rpc-url provided".to_string())?,
```

Comment:
I would make this a toplevel function or even like `from_args` on `Config`, it doesn't really make sense to be a member function on `Args`.  This also makes some sense to use `anyhow` instead of manipulating string error messages.

Reply:
The idea behind this was that Args seems to be more inside the bin whereas Config seems somewhere common which other crates can/do use. To me, it doesn't make sense to pull in Arg dependency where config is defined.

I think config should not be in bin. It is just being used everywhere so it makes sense to keep it somewhere common and it should not have dependency on Args.

Reply:
If we're separating the actual *service* config from the inner *client*/*node* config internally then yeah moving it apart makes sense.

But how these fns are organized here does feel weird to me.